### ISSUE TITLE
Error for prediction on large sparse matrix

### ISSUE THREAD

**Issue by FelixNeutatz** [dmlc/xgboost#4492]
Dear all,

I just encountered the following exception while predicting on a very large sparse matrix:
```
    probability_prediction_all = self.model.predict_proba(self.X_train)
  File "/usr/local/lib/python2.7/dist-packages/xgboost/sklearn.py", line 810, in predict_proba
    test_dmatrix = DMatrix(data, missing=self.missing, nthread=self.n_jobs)
  File "/usr/local/lib/python2.7/dist-packages/xgboost/core.py", line 403, in __init__
    self._init_from_csr(data)
  File "/usr/local/lib/python2.7/dist-packages/xgboost/core.py", line 440, in _init_from_csr
    c_array(ctypes.c_uint, csr.indices),
  File "/usr/local/lib/python2.7/dist-packages/xgboost/core.py", line 219, in c_array
    return (ctype * len(values))(*values)
SystemError: ../Objects/tupleobject.c:54: bad argument to internal function
Exception xgboost.core.XGBoostError: XGBoostError('[22:53:24] /workspace/src/c_api/c_api.cc:756: DMatrix/Booster has not been intialized or has already been disposed.\n\nStack trace returned 10 entries:\n[bt] (0) /usr/local/lib/python2.7/dist-packages/xgboost/./lib/libxgboost.so(dmlc::StackTrace(unsigned long)+0x47) [0x7f00bc4f7fc7]\n[bt] (1) /usr/local/lib/python2.7/dist-packages/xgboost/./lib/libxgboost.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x1d) [0x7f00bc4f842d]\n[bt] (2) /usr/local/lib/python2.7/dist-packages/xgboost/./lib/libxgboost.so(XGDMatrixFree+0xc6) [0x7f00bc4ed8e6]\n[bt] (3) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call_unix64+0x4c) [0x7f00d8dc5e40]\n[bt] (4) /usr/lib/x86_64-linux-gnu/libffi.so.6(ffi_call+0x2eb) [0x7f00d8dc58ab]\n[bt] (5) /usr/lib/python2.7/lib-dynload/_ctypes.x86_64-linux-gnu.so(_ctypes_callproc+0x48f) [0x7f00d8fd53df]\n[bt] (6) /usr/lib/python2.7/lib-dynload/_ctypes.x86_64-linux-gnu.so(+0x11d82) [0x7f00d8fd9d82]\n[bt] (7) python(PyEval_EvalFrameEx+0x578d) [0x4c166d]\n[bt] (8) python(PyEval_EvalCodeEx+0x306) [0x4b9b66]\n[bt] (9) python() [0x4d5669]\n\n',) in <bound method DMatrix.__del__ of <xgboost.core.DMatrix object at 0x7f00a3de7f90>> ignored
```

```
Python 2.7.12
xgboost 0.82
```

Is this a known issue?

For now, I solve the issue by using the .todense() method on the matrix.

Best regards,
Felix

**Comment by trivialfis** [comment]
@FelixNeutatz Never seen it before.  It seems the error is coming from Python itself.  The dmatrix handling error is a side effect of previous error and is fixed in 0.90.(but no more Python 2 support in 0.90)

**Comment by FelixNeutatz** [comment]
Thank you @trivialfis for the quick answer :)
